components/eam/src/physics/crm/crm_history.F90

```diff
@@ -443,126 +447,129 @@ subroutine crm_history_out(state, ptend, crm_state, crm_rad, crm_output, crm_ecp
    ! CRM_QRS + CRM_QRL output in radiation_tend will show a time lag.
 
    ! GCM level rad heating tendencies
-   call outfld('MMF_QRL   ',qrl/cpair, pcols, lchnk )
-   call outfld('MMF_QRS   ',qrs/cpair, pcols, lchnk )
+   call outfld('MMF_QRL   ',qrl/cpair, ncol, lchnk )
+   call outfld('MMF_QRS   ',qrs/cpair, ncol, lchnk )
 
    ! Why do we output this here?
    call outfld('PRES    ',state%pmid, pcols, lchnk )
    call outfld('DPRES   ',state%pdel, pcols, lchnk )
 
    ! CRM state variables on CRM grid
-   call outfld('CRM_U   ',crm_state%u_wind,      pcols, lchnk )
-   call outfld('CRM_V   ',crm_state%v_wind,      pcols, lchnk )
-   call outfld('CRM_W   ',crm_state%w_wind,      pcols, lchnk )
-   call outfld('CRM_T   ',crm_state%temperature, pcols, lchnk )
+   call outfld('CRM_U   ',crm_state%u_wind     (icrm_beg:icrm_end,:,:,:), ncol, lchnk )
```

Comment:
I think this might be a little dangerous, we might want to add an assertion to make sure that ncol = icrm_end - icrm_beg + 1.

Reply:
This is guaranteed by how the icrm values are set in the above routine, but it wouldn't hurt to have an extra check here.

Reply:
Added check in latest commits